### DISCUSSION TITLE
Ideas on how to not include connection string in JSON file

### DISCUSSION THREAD

**Discussion by roguepixelation** [joyfullservice/msaccess-vcs-addin#475] [Q&A] [answered]
Hi everyone, 
I want to start versioning my now exported access file, but I don't necessarily want to include the connection string in the table JSON file. Is it possible to somehow have the connection string in a different file, so that that file can be added to the gitignore, but also be able to build from source again? (if the connection string file is provided)

**Comment by joyfullservice** [comment, marked as answer]
That's a great question! In many corporate environments a linked table may use Active Directory integrated authentication, so no sensitive credential information is stored in the source files. But for other types of databases or authentication strategies, there IS sensitive information stored in these connection strings, and you are exactly right, that these really should not be committed to version control.

The good news is that a few months ago I added support for using `.env` files to store sensitive credentials for external database connections. (See #415) This functionality could probably be extended for use with linked tables (and pass-through queries, and other objects using a connection string.)

I would suggest that you create a new issue as a suggested enhancement for this add-in, and we can collaborate the development discussion around the issue.

**Reply by roguepixelation** [reply]
Hi Adam, 
that sounds great. I'll go ahead and create the issue.

Thanks!
